### ISSUE TITLE
🐛 [SQL] Trial of the Magian CharVar issue "Multi Process"

### ISSUE THREAD

**Issue by N3ckB3ard** [DerpyProjectGroup/topaz#3615]
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

 Windows 10 (x64- Debug)

## Branch affected by issue 

`base`
![image](https://user-images.githubusercontent.com/98081508/219997893-f31e3fec-49b8-4e03-a614-d83f9833a66f.png)

## Steps to reproduce

I am currently experiencing a issue on CatsEyeXI regarding Trial of the Magian. It seems and through testing Trials are being accounted for correctly for objective total and count. Then when the trial is complete if you are to return to the moogle to turn in your progress or completed item you will be informed your trial is not completed.

Printing some of the local's within
https://github.com/LandSandBoat/server/blob/70680538d6c2a6fec0e960cd5407fbc0df142d5c/scripts/globals/magiantrials.lua#L21
I then adjusted this return to look for the charVar [trial]2

gave me a odd return...
![image](https://user-images.githubusercontent.com/98081508/219922118-89320352-24c8-40d6-ace0-31eecf448550.png)

```
!checkvar Neckbeard [trial]2
```
Returns the correct Variable for the trial

It seems more than this global is being effected if the getCharVar can not return the value from the DB. Thank you for anyone with information regarding this.

## Update 2/19:
I was successfully able to recreate this issue locally and it exhibits the same behaviors as the issue from CatsEyeXI.
Unfortunately the SQL log error seems to be specific to the DB of the server so I am removing most of the issues related to it.

For my test today I enabled logging as suggested from Zach and updated to the latest build. I then started my multiple processes of map.exe targeting different ports.
(Note: Ru'Lude Gardens is **NOT** on port.54230)

Checking my trial by trading it to the moogle should reflect my current progression of completed trial objectives.
![image](https://user-images.githubusercontent.com/98081508/220001227-dbb451de-390a-4a9e-9f4d-2b3b455043d9.png)
![image](https://user-images.githubusercontent.com/98081508/220005491-44c072a7-89fd-41ef-a751-45bae60d9c0c.png)

After completing another objective I returned to the moogle to find my trial progress remains the same even if my CharVariable
increases.
![image](https://user-images.githubusercontent.com/98081508/220001442-bea1ace8-1082-4f56-902b-efcb0c40ab1d.png)

I then added the following prints to my console to see if I could collect the data that is being used during this NPC interaction. 
https://github.com/LandSandBoat/server/blob/70680538d6c2a6fec0e960cd5407fbc0df142d5c/scripts/globals/magiantrials.lua#L454-L455
```lua
                local charVar      = player:getCharVar("[trial]2")
                print("_, cacheData:", _, cacheData)
                print("progress:", progress)
                print("itemId:", itemId)
                print("charVar:", charVar)
```
this was my return
![image](https://user-images.githubusercontent.com/98081508/220001628-4000c6f7-5058-44ff-af2b-99f24b902671.png)
Then after I restarted the process (map.exe that matched the port to Ru'Lude Gardens) my CharVar reflected correctly and I received the correct progress.
![image](https://user-images.githubusercontent.com/98081508/220002920-a88603b9-6bb7-47ac-b716-49dd8ec7912e.png)

This ultimately prevents players from being able to turn in completed trial's as the CharVar I imagine is being set when the exe loads the area and somehow is not being updated when the CharVar increases.
## Expected behavior

Return of the correct CharVar ~~and solution to SQL errors~~

When you have completed the total amount of objectives related to your trial and redeem your reward without a restart of the map process

**Comment by N3ckB3ard** [comment]
![image](https://user-images.githubusercontent.com/98081508/219923008-694c983e-1f1a-4f1c-8acf-7335045bb66d.png)
More logged errors

**Comment by zach2good** [comment]
Any more context? I can't repro with 7x process MP:
<img width="683" alt="image" src="https://user-images.githubusercontent.com/1389729/219942905-45c97b85-401a-4f6d-be0b-705ceaecac3a.png">

**Comment by zach2good** [comment]
I'm in the process of adding some more context to that logging: https://github.com/LandSandBoat/server/pull/3617

But in the meantime you can turn on `DEBUG_SQL` and see what queries are being sent by the server: https://github.com/LandSandBoat/server/blob/base/settings/default/logging.lua#L52
https://github.com/LandSandBoat/server/blob/base/src/common/sql.cpp#L300

It'll be noisy, but as long as you're doing this on your local one-man server, it shouldn't be too bad

**Comment by N3ckB3ard** [comment]
> Any more context? I can't repro with 7x process MP: <img alt="image" width="683" src="https://user-images.githubusercontent.com/1389729/219942905-45c97b85-401a-4f6d-be0b-705ceaecac3a.png">

I can confirm not a single process had access to SQL almost as if the single thread is being locked? As of right now I did notice all players with this issue seem to go away after a complete server reboot. This issue begins to happen after a undisclosed amount of time

**Comment by N3ckB3ard** [comment]
![image](https://user-images.githubusercontent.com/98081508/219988631-7b87ace6-9829-4985-8557-5ff0663e2521.png)
Update I was able to reproduce this on base LSB using the same cluster style as the server. I believe the sql error to now be removed as a related issue?

**Comment by CatsEyeXI** [comment]
We merged LSB today to grab #3617. This is what I found... most seem to be related to inserting LS messages in the audit chat.

![image](https://user-images.githubusercontent.com/65316697/220691515-f3f70bdf-f3e0-415f-8eb9-49873616702f.png)

**Comment by zach2good** [comment]
Ah, I see it, Async worker thread is created on the main thread:
https://github.com/LandSandBoat/server/blob/base/src/common/async.cpp#L27-L55

Needs to be created and managed by the worker. Free PR for anyone who wants it.
